### PR TITLE
Add caching for pending proposals

### PR DESCRIPTION
**Pull request by toninorair**
An approach that allows:
- fetching pending proposals once inside worker
- caching them 
- reading this info from cache for all scenarios.

### CODE REVIEW

**Review by jflatow — COMMENTED** [review]
So this assumes the slow part is getting all pending proposals? I tend to think its the running the migrations/executing them. Want to try on the mainnet proposal branch to benchmark?

**Comment by antonina-cherednichenko** [comment]
> So this assumes the slow part is getting all pending proposals? I tend to think its the running the migrations/executing them. Want to try on the mainnet proposal branch to benchmark?

About 2 min difference

File: plugins/scenario/worker/Worker.ts [hunk 1]
```diff
@@ -62,6 +63,11 @@ export async function run<T, U, R>({ bases, config, worker }: WorkerData) {
   for (const base of bases) {
     const world = new World(hreForBase(base), base), dm = world.deploymentManager;
     const delta = await dm.runDeployScript({ allMissing: true });
+
+    // Reading pending proposals and storing them in the cache for all scenarios
+    const proposals = await getAllPendingProposals(world);
+    dm.cache.storeCache(getProposalCacheId(base.network), proposals);
```
Comment (jflatow) [inline plugins/scenario/worker/Worker.ts:69]: The cache is stored across networks, so not sure we would want to do this without accounting for forking. Also, the plugin code is meant to be agnostic to Compound-specific stuff, so we wouldn't want it to be aware of proposals, right?

**Comment by jflatow** [comment]
Closing as this is better addressed by #700

**Pull request closed without merging**